internal/opts/host_gateway_test.go

```diff
@@ -0,0 +1,98 @@
+package opts
+
+import (
+	"encoding/json"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+)
+
+func TestHostGateway(t *testing.T) {
+	testcases := []struct {
+		name   string
+		input  []string
+		expStr string
+		expErr string
+	}{
+		{
+			name:   "ipv4",
+			input:  []string{"10.1.1.1"},
+			expStr: "10.1.1.1",
+		},
+		{
+			name:   "ipv6",
+			input:  []string{"fdb8:3037:2cb2::1"},
+			expStr: "fdb8:3037:2cb2::1",
+		},
+		{
+			name:   "ipv6 and ipv4",
+			input:  []string{"fdb8:3037:2cb2::1", "10.1.1.1"},
+			expStr: "10.1.1.1,fdb8:3037:2cb2::1",
```

Comment:
Shouldn't it be a JSON array?
```suggestion
			expStr: `["10.1.1.1","fdb8:3037:2cb2::1"]`,
```

Reply:
It'll never be used as JSON, so I went for something unambiguous and minimal for error messages etc. At-least `opts/address_pools.go` takes the same approach.

The two addresses will need to be passed to buildx, once it understands two addresses. Either version would be ok for that.

Can change it if needed though?

Reply:
Was mostly concerned if it should match the input in `daemon.json` it was parsed from.
If it doesn't matter though, either is fine.